cmd/dep/status.go

```diff
@@ -150,6 +158,123 @@ func (out *jsonOutput) MissingFooter() {
 	json.NewEncoder(out.w).Encode(out.missing)
 }
 
+type dotProject struct {
+	project  string
+	version  string
+	children []string
+}
+
+func (dp *dotProject) hash() uint32 {
+	h := fnv.New32a()
+	h.Write([]byte(dp.project))
+	return h.Sum32()
+}
+
+func (dp *dotProject) label() string {
+	label := []string{dp.project}
+
+	if dp.version != "" {
+		label = append(label, dp.version)
+	}
+
+	return strings.Join(label, "\n")
+}
+
+type dotOutput struct {
+	w   io.Writer
+	o   string
+	p   *dep.Project
+	dps []*dotProject
+	b   bytes.Buffer
+	bsh map[string]uint32
+}
+
+func (out *dotOutput) BasicHeader() {
+
+	// TODO Check for dot package before doing something
+	// cmd := exec.Command("dot", "-V")
+
+	out.dps = []*dotProject{}
+	out.bsh = make(map[string]uint32)
+	out.b.WriteString("digraph { node [shape=box]; ")
+
+	ptree, _ := gps.ListPackages(out.p.AbsRoot, string(out.p.ImportRoot))
+	prm, _ := ptree.ToReachMap(true, false, false, nil)
+
+	rdp := &dotProject{
+		project:  string(out.p.ImportRoot),
+		version:  "",
+		children: prm.Flatten(false),
+	}
+
+	out.dps = append(out.dps, rdp)
+}
+
+func (out *dotOutput) BasicFooter() {
+
+	for _, dp := range out.dps {
+		out.bsh[dp.project] = dp.hash()
+
+		// Create name boxes, and name them using hashes
+		// to avoid encoding name conflicts
```

Comment:
nit: either use just a couple clarifying words, or a complete sentence (including a period) in comments, please